MacOS Catalina + Python 3.8 - Failing to build wrappers with binaries installed via brew

This is the output after running : python3 setup.py install

running install
running build
running build_ext
skipping 'freenect.c' Cython extension (up-to-date)
building 'freenect' extension
creating build
creating build/temp.macosx-10.9-x86_64-3.8
gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch x86_64 -g -I/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8 -c freenect.c -o build/temp.macosx-10.9-x86_64-3.8/freenect.o -fPIC -I ../../include/ -I /usr/include/libusb-1.0/ -I /usr/local/include/libusb-1.0 -I /usr/local/include -I ../c_sync/ -I /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/numpy/core/include
In file included from freenect.c:353:
In file included from /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/numpy/core/include/numpy/arrayobject.h:4:
In file included from /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/numpy/core/include/numpy/ndarrayobject.h:12:
In file included from /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/numpy/core/include/numpy/ndarraytypes.h:1830:
/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: 
      "Using deprecated NumPy API, disable it with "          "#define
      NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-W#warnings]
#warning "Using deprecated NumPy API, disable it with " \
 ^
freenect.c:8444:134: warning: code will never be executed [-Wunreachable-code]
  ...__pyx_t_2++; if (unlikely(0 < 0)) {__pyx_filename = __pyx_f[1]; __pyx_li...
                                                         ^~~~~~~
freenect.c:10210:31: warning: 'tp_print' is deprecated
      [-Wdeprecated-declarations]
  __pyx_type_8freenect_CtxPtr.tp_print = 0;
                              ^
/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/cpython/object.h:260:5: note: 
      'tp_print' has been explicitly marked deprecated here
    Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
    ^
/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/pyport.h:515:54: note: 
      expanded from macro 'Py_DEPRECATED'
#define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                     ^
freenect.c:10214:31: warning: 'tp_print' is deprecated
      [-Wdeprecated-declarations]
  __pyx_type_8freenect_DevPtr.tp_print = 0;
                              ^
/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/cpython/object.h:260:5: note: 
      'tp_print' has been explicitly marked deprecated here
    Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
    ^
/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/pyport.h:515:54: note: 
      expanded from macro 'Py_DEPRECATED'
#define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                     ^
freenect.c:10218:33: warning: 'tp_print' is deprecated
      [-Wdeprecated-declarations]
  __pyx_type_8freenect_StatePtr.tp_print = 0;
                                ^
/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/cpython/object.h:260:5: note: 
      'tp_print' has been explicitly marked deprecated here
    Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
    ^
/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/pyport.h:515:54: note: 
      expanded from macro 'Py_DEPRECATED'
#define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                     ^
freenect.c:11588:21: error: no member named 'exc_type' in 'struct _ts'
    *type = tstate->exc_type;
            ~~~~~~  ^
freenect.c:11589:22: error: no member named 'exc_value' in 'struct _ts'; did you
      mean 'curexc_value'?
    *value = tstate->exc_value;
                     ^~~~~~~~~
                     curexc_value
/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/cpython/pystate.h:79:15: note: 
      'curexc_value' declared here
    PyObject *curexc_value;
              ^
freenect.c:11590:19: error: no member named 'exc_traceback' in 'struct _ts'; did
      you mean 'curexc_traceback'?
    *tb = tstate->exc_traceback;
                  ^~~~~~~~~~~~~
                  curexc_traceback
/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/cpython/pystate.h:80:15: note: 
      'curexc_traceback' declared here
    PyObject *curexc_traceback;
              ^
freenect.c:11602:24: error: no member named 'exc_type' in 'struct _ts'
    tmp_type = tstate->exc_type;
               ~~~~~~  ^
freenect.c:11603:25: error: no member named 'exc_value' in 'struct _ts'; did you
      mean 'curexc_value'?
    tmp_value = tstate->exc_value;
                        ^~~~~~~~~
                        curexc_value
/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/cpython/pystate.h:79:15: note: 
      'curexc_value' declared here
    PyObject *curexc_value;
              ^
freenect.c:11604:22: error: no member named 'exc_traceback' in 'struct _ts'; did
      you mean 'curexc_traceback'?
    tmp_tb = tstate->exc_traceback;
                     ^~~~~~~~~~~~~
                     curexc_traceback
/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/cpython/pystate.h:80:15: note: 
      'curexc_traceback' declared here
    PyObject *curexc_traceback;
              ^
freenect.c:11605:13: error: no member named 'exc_type' in 'struct _ts'
    tstate->exc_type = type;
    ~~~~~~  ^
freenect.c:11606:13: error: no member named 'exc_value' in 'struct _ts'; did you
      mean 'curexc_value'?
    tstate->exc_value = value;
            ^~~~~~~~~
            curexc_value
/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/cpython/pystate.h:79:15: note: 
      'curexc_value' declared here
    PyObject *curexc_value;
              ^
freenect.c:11607:13: error: no member named 'exc_traceback' in 'struct _ts'; did
      you mean 'curexc_traceback'?
    tstate->exc_traceback = tb;
            ^~~~~~~~~~~~~
            curexc_traceback
/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8/cpython/pystate.h:80:15: note: 
      'curexc_traceback' declared here
    PyObject *curexc_traceback;
              ^
5 warnings and 9 errors generated.
error: command 'gcc' failed with exit status 1

--- a/wrappers/python/freenect.c
+++ b/wrappers/python/freenect.c
@@ -11585,9 +11585,15 @@ static void __Pyx_WriteUnraisable(const char *name, CYTHON_UNUSED int clineno,
 static CYTHON_INLINE void __Pyx_ExceptionSave(PyObject **type, PyObject **value, PyObject **tb) {
 #if CYTHON_COMPILING_IN_CPYTHON
     PyThreadState *tstate = PyThreadState_GET();
+    #if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 7
+    *type = tstate->exc_state.exc_type;
+    *value = tstate->exc_state.exc_value;
+    *tb = tstate->exc_state.exc_traceback;
+    #else
     *type = tstate->exc_type;
     *value = tstate->exc_value;
     *tb = tstate->exc_traceback;
+    #endif
     Py_XINCREF(*type);
     Py_XINCREF(*value);
     Py_XINCREF(*tb);
@@ -11599,12 +11605,23 @@ static void __Pyx_ExceptionReset(PyObject *type, PyObject *value, PyObject *tb) 
 #if CYTHON_COMPILING_IN_CPYTHON
     PyObject *tmp_type, *tmp_value, *tmp_tb;
     PyThreadState *tstate = PyThreadState_GET();
+    #if PY_MAJOR_VERSION >= 3 && PY_MINOR_VERSION >= 7
+    tmp_type = tstate->exc_state.exc_type;
+    tmp_value = tstate->exc_state.exc_value;
+    tmp_tb = tstate->exc_state.exc_traceback;
+    tstate->exc_state.exc_type = type;
+    tstate->exc_state.exc_value = value;
+    tstate->exc_state.exc_traceback = tb;
+    #else
     tmp_type = tstate->exc_type;
     tmp_value = tstate->exc_value;
     tmp_tb = tstate->exc_traceback;
     tstate->exc_type = type;
     tstate->exc_value = value;
     tstate->exc_traceback = tb;
+    #endif
+    
+   
     Py_XDECREF(tmp_type);
     Py_XDECREF(tmp_value);
     Py_XDECREF(tmp_tb);

